src/app/hooks/useNetworkFees.ts

```diff
@@ -25,39 +34,62 @@ const formatFee = (fee: string, rate: BigNumber) => {
     fiat: CurrencyFormatter.cryptoToCurrency(cryptoAmount, rate, {
       decimals: 2,
     }),
-    crypto: cryptoAmount.decimalPlaces(4).toFixed(4),
+    crypto: Number.parseFloat(
+      cryptoAmount.decimalPlaces(4).toFixed(4),
+    ).toString(),
   };
 };
 
-export const useNetworkFees = (network: NetworkType) => {
-  const networkUrls = {
+export const useNetworkFees = (network: NetworkType, type: TransactionType) => {
+  const dynamicFeeNetworkUrls = {
     [NetworkType.DEVNET]: "https://dwallets.ark.io/api/node/fees",
     [NetworkType.MAINNET]: "https://wallets.ark.io/api/node/fees",
   };
 
-  const { data: fees } = useQuery({
-    queryKey: ["network-fees", network],
+  const staticFeeNetworksUrls = {
+    [NetworkType.DEVNET]: "https://dwallets.ark.io/api/transactions/fees",
+    [NetworkType.MAINNET]: "https://wallets.ark.io/api/transactions/fees",
+  };
+
+  const { data: dynamicFees } = useQuery({
+    queryKey: ["dynamic-network-fees", network],
+    staleTime: 0,
+    refetchInterval: 3 * 60 * 1000, // 3 minutes
+    queryFn: async () => {
+      const jsonResponse = await fetch(dynamicFeeNetworkUrls[network]);
+      const response = (await jsonResponse.json()) as DynamicFeesApiResponse;
+
+      return type === TransactionType.VOTE
+        ? response.data["1"].vote
+        : response.data["1"].transfer;
```

Comment:
I'd keep the whole API response cached and extract bits I need later, this is essential as the query key is `["dynamic-network-fees", network]`. You either add tx type to the query key array or fix the return valur. I'd go with return value fix as it'd prevent possible get requests.